packages/telemetry/src/slog-file.js

```diff
@@ -1,65 +1,21 @@
-import { createWriteStream } from 'fs';
-import { open } from 'fs/promises';
-import { E } from '@endo/eventual-send';
-import { fsStreamReady, makeAggregateError } from '@agoric/internal';
+import { makeFsStreamWriter } from '@agoric/internal';
 import { serializeSlogObj } from './serialize-slog-obj.js';
 
-const noPath = /** @type {import('fs').PathLike} */ (
-  /** @type {unknown} */ (undefined)
-);
-
 /** @param {import('./index.js').MakeSlogSenderOptions} opts */
 export const makeSlogSender = async ({ env: { SLOGFILE } = {} } = {}) => {
-  if (!SLOGFILE) {
+  const stream = await makeFsStreamWriter(SLOGFILE);
+
+  if (!stream) {
     return undefined;
   }
 
-  const handle = await open(SLOGFILE, 'a');
-
-  const stream = createWriteStream(noPath, { fd: handle.fd });
-  await fsStreamReady(stream);
-
-  let flushed = Promise.resolve();
-
-  const writeNewLine = () => {
-    /** @type {Promise<void>} */
-    const written = new Promise((resolve, reject) => {
-      stream.write('\n', err => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve();
-        }
-      });
-    });
-    flushed = E.when(
-      flushed,
-      () => written,
-      async err =>
-        Promise.reject(
-          written.then(
-            () => err,
-            writtenError => makeAggregateError([err, writtenError]),
-          ),
-        ),
-    );
-  };
-
   const slogSender = (slogObj, jsonObj = serializeSlogObj(slogObj)) => {
-    stream.write(jsonObj);
-    writeNewLine();
+    // eslint-disable-next-line prefer-template
+    void stream.write(jsonObj + '\n');
```

Comment:
This technically causes a single write instead of 2, which may be observable if the SLOGFILE is a pipe (which the loadgen uses, and will actually improve perf).